finality-aleph/src/metrics.rs

```diff
@@ -84,6 +90,50 @@ impl BlockMetrics {
         Self::Noop
     }
 
+    /// Updates metrics for `checkpoint` so that all blocks referenced previously with `header_hash`
+    /// are now referenced with `post_hash`. This is useful when a block is first reported
+    /// with header hash not including post-digests, and later it is reported with header hash
+    /// including post digests.
```

Comment:
Hm, we haven't done that previously, was this just a bug? Also, this currently only happens in `TracingBlockImport`, not in `BlockImporter`, which means it will not happen in the future? Also also, I think finalization uses the non-post version, so this won't work if the hashes ever differ?

I might be completely misunderstanding what is going on here though. I _think_ we currently always have `hash == post_hash`, which means none of this matters for now, but the code definitely doesn't look correct in general. :/

Reply:
You were right, we are using only `post_hash`, so this part of code was not needed and incorrect.